packages/panzer/dof-mgr/src/Panzer_DOFManager_impl.hpp

```diff
@@ -789,13 +789,9 @@ DOFManager<LO,GO>::buildGlobalUnknowns_GUN(const Tpetra::MultiVector<GO,LO,GO,pa
   GO localsum=0;
   {
     PANZER_DOFMGR_FUNC_TIME_MONITOR("panzer::DOFManager::buildGlobalUnknowns_GUN::line_07-09 local_count");
-
-    typedef typename Tpetra::MultiVector<GO,Node> MV;
-    typedef typename MV::dual_view_type::t_dev KV;
-    typedef typename MV::dual_view_type::t_dev::memory_space DMS;
-    KV values = non_overlap_mv->template getLocalView<DMS>();
+    auto values = non_overlap_mv->getLocalViewDevice();
     auto mv_size = values.extent(0);
-    Kokkos::parallel_reduce(mv_size,panzer::dof_functors::SumRank2<GO,KV>(values),localsum);
+    Kokkos::parallel_reduce(mv_size,panzer::dof_functors::SumRank2<GO,decltype(values)>(values),localsum);
```

Comment:
In the future, please consider using an explicit RangePolicy instead of putting the dimension as the first argument of `Kokkos::parallel_reduce`.  This would let you execute on the `execution_space` corresponding to Tpetra's data, rather than on Kokkos' default execution space.  Tpetra's default execution space could differ from Kokkos' default execution space.  (Some people really want Serial for Tpetra but CUDA for everything else.)